frappe/client.py

```diff
@@ -312,3 +313,37 @@ def get_js(items):
 def get_time_zone():
 	'''Returns default time zone'''
 	return {"time_zone": frappe.defaults.get_defaults().get("time_zone")}
+
+@frappe.whitelist()
+def attach_file():
+	'''Attach a file to Document (POST)
+
+	:param filename: filename e.g. test-file.txt
+	:param filedata: base64 encode filedata
+	:param doctype: Reference DocType to attach file
+	:param docname: Reference DocName to attach file
+	:param folder: Folder to add File into
+	:param decode: decode filedata from base64 encode
+	:param is_private: Attach file as private file
+	:param docfield: file to attach to'''
+
+	filename = frappe.form_dict.get("filename")
+	filedata = frappe.form_dict.get("filedata")
+	doctype = frappe.form_dict.get("doctype")
+	docname = frappe.form_dict.get("docname")
+	folder = frappe.form_dict.get("folder")
+	decode = frappe.form_dict.get("decode")
```

Comment:
what is decode (?) "base64" ?

Reply:
Yes. It is the Param required by save_file(). 

If decode is set to false, base64 encoded string is saved as text file.

If it is set to true like example in PR opening post. The base64 encoded string will be decoded into appropriate file.